### PR TITLE
Dala 1394 linting precommithook

### PR DESCRIPTION
**Pull request by ahoecherl**
# Set up pre-commit hook
Pre commit hook does:

- Run ktLint Format over all changed, staged files.
- Run detekt over all files (since detekt also checks interdependencies of files, i.e. would be inclomplete if only run for changed files
- Run eslint on all changes .ts and .vue files

To actually use the pre-commit hook you need to change your local git config as documented in the updated README.md

## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [ ] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server "dev1" using this branch
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma

### CODE REVIEW

File: .githooks/pre-commit
Comment (fsoffel) [inline .githooks/pre-commit:1]: There is no file ending on this file.

File: README.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ In this section, you find information that might be useful for you as a develope
 Especially under Windows, it's unclear which file permissions a script will get. 
 To explicitly mark a script executable, do:
 `git update-index --chmod=+x script.sh`
+## Git Hooks
+To add the provided git pre-hooks to your local development environment execute:
+`git config --local core.hooksPath ./githooks/`
```
Comment (fsoffel) [inline README.md:24]: it should be ./.githooks

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ./gradlew ktlintFormat -PinternalKtlintGitFilter="$KT_FILES"
+
+  echo "Completed ktlint run."
+
+  echo "Starting detekt run"
+  ./gradlew detekt
+  echo "Completed detekt run"
+
+  echo "$KT_FILES" | while read -r file; do
+      if [ -f $file ]; then
```
Comment (fsoffel) [inline .githooks/pre-commit:22]: Why is this extra check neccessary?

File: .githooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          git add $file
```
Comment (fsoffel) [inline .githooks/pre-commit:23]: Files get commited if this is removed, too. So why is this needed?

Comment (ahoecherl) [reply .githooks/pre-commit:23]: re-adding the file is necessary to make the changes done by kt-lint part of the ongoing commit. Otherwise just  the previously staged status of the code (pre-kt-lint) is committed and an additional manually executed commit would be necessary to actually commit the kt-lint changes.

File: .githooks/pre-commit (hunk 2 above)
Comment (ahoecherl) [reply .githooks/pre-commit:22]: The if checks if a file does actually exist before adding it via git.

I'm not entirely sure which in which edge-case this would become relevant. I was of course "heavily inspired" from something I found on the internet when writing this hook. If for some reason ktlintFormat has entirely deleted a file this if statement would become relevant.

I would prefer to leave it in just in case it does actually does cover some edge case.

File: .githooks/pre-commit [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  ./gradlew ktlintFormat -PinternalKtlintGitFilter="$KT_FILES"
+
+  echo "Completed ktlint run."
+
+  echo "Starting detekt run"
+  ./gradlew detekt
+  echo "Completed detekt run"
+
+  echo "$KT_FILES" | while read -r file; do
```
[the remark this answers is not in the document]
Comment (ahoecherl) [reply .githooks/pre-commit:21]: Tried 

`for file in "$EDIT_FILES"
do
  if [ -f $file ]; then
    git add $file
  fi
done
`

instead but it didn't work. Since the while loop does what we need I'd propose to just stick with it.

File: .githooks/pre-commit (hunk 2 above)
Comment (fsoffel) [reply .githooks/pre-commit:22]: 😄 Ok, it doesn't hurt in any case.

File: .githooks/pre-commit (hunk 4 above)
[the remark this answers is not in the document]
Comment (fsoffel) [reply .githooks/pre-commit:21]: As discussed, the while loop is perfectly fine.
